web/php/Actions/PMAction.php

```diff
@@ -1,326 +1,218 @@
 
-        $db = Database::connection();
-        $db->begin();
+        $message->delete();
 
-        $c2 = new Criteria();
-        foreach ($selected as $s) {
-            $c2->addOr("message_id", $s);
+        if ($nextMessage) {
+            $runData->ajaxResponseAdd('messageId', $nextMessage->id);
         }
-        $c->addCriteriaAnd($c2);
-
-        PrivateMessagePeer::instance()->delete($c);
-
-        $db->commit();
     }
 
-    public function removeDraftsMessageEvent($runData)
+    /**
+     * Remove selected items from the user's Sent folder.
+     * @param RunData $runData
+     */
+    public function removeSelectedSentEvent(RunData $runData)
     {
-        $messageId = $runData->getParameterList()->getParameterValue("message_id");
-        $userId = $runData->getUserId();
-
-        $db = Database::connection();
-        $db->begin();
-
-        $c = new Criteria();
-        $c->add("message_id", $messageId);
-        $c->add("from_user_id", $userId);
-        $c->add("flag", 2);
-
-        PrivateMessagePeer::instance()->delete($c);
-        $c = new Criteria();
-        $c->add("from_user_id", $userId);
-        $c->add("message_id", $messageId, "<");
-        $c->add("flag", 2);
-        $c->addOrderDescending("message_id");
-
-        $mid = PrivateMessagePeer::instance()->selectOne($c);
-        if ($mid == null) {
-                $c = new Criteria();
-            $c->add("from_user_id", $userId);
-            $c->add("message_id", $messageId, ">");
-            $c->add("flag", 2);
-            $c->addOrderAscending("message_id");
-
-            $mid = PrivateMessagePeer::instance()->selectOne($c);
-        }
-
-        if ($mid !== null) {
-            $runData->ajaxResponseAdd("messageId", $mid->getMessageId());
-        }
+        $selected = $runData->get('selected');
+        $messages = (new JSONService(SERVICES_JSON_LOOSE_TYPE))->decode($selected);
 
-        $db->commit();
+        UserMessage::sent($runData->user())
+            ->whereIn('id', $messages)
+            ->delete();
     }
 
-    public function removeSelectedDraftsEvent($runData)
+    /**
+     * Delete a single draft and present the next one to the user.
+     * @param RunData $runData
+     */
+    public function removeDraftsMessageEvent(RunData $runData)
     {
-        $userId = $runData->getUserId();
-        $c = new Criteria();
-        $c->add("from_user_id", $userId);
-        $c->add("flag", 2);
+        $message = UserMessage::find($runData->get('message_id'));
 
-        $selected = $runData->getParameterList()->getParameterValue("selected");
-        $json = new JSONService(SERVICES_JSON_LOOSE_TYPE);
-        $selected = $json->decode($selected);
+        $nextMessage = UserMessage::drafts($runData->user())
+                ->whereTime('id', '<', $message->created_at)->first()
+            ??
+            UserMessage::drafts($runData->user())
+                ->whereTime('id', '>', $message->created_at)->first();
```

Comment:
I feel like this is non-obvious _enough_ that it's it's worth making into an explicit if statement instead of null coalesce. Or leave a comment explaining why this is happening I guess.

Alternatively, make an inline helper function that takes an argument `'<'` or `'>'` and then call `helperNameHere('<') ?? helperNameHere('>') // Get the next message, or previous if it's the last one`